priority_que.py

```diff
@@ -0,0 +1,41 @@
+import bisect
+
+from binheap import BinaryHeap
+
+
+class PriorityQue:
+    def __init__(self, values=None):
+        self._storage = BinaryHeap("max")
+        if values:
+            for priority, value in values:
+                self._storage.push(Node(priority, value))
+
+    def __len__(self):
+        return self._storage._size
+
+    def insert(self, value, priority=0):
+        self._storage.push(Node(priority, value))
+
+    def pop(self):
+        try:
+            return self._storage.pop().value
+        except IndexError:
+            raise IndexError("No items to pop.")
+
+    def peek(self):
+        try:
+            return self._storage._storage[0].value
+        except IndexError:
+            raise IndexError("No items to see.")
```

Comment:
While catching the Index error here is Okay, Queues don't have Index, so throwing an Index Error isn't correct.